### PR TITLE
Fix issues with STJ serializer

### PR DESCRIPTION
**Pull request by VilleHakli**
- Serializing all details for triggers that AdoJobStore stores
- Improve JSON serialization tests
- Add Verify & TimeProvider fake to make testing easier
- Remove outdated comments about JsonPropertyAttribute
- Fix TODOs in tests related to STJ serialization which were added in PR #2435

#2426

### CODE REVIEW

**Comment by VilleHakli** [comment]
This has all the same changes as in #2434 but this time targeting main. Also contains fixes for TODOs added in #2435

Verify.NUnit adds implicit using for NUnit.Framework and as main has implicit usings enabled, IDE shows a lot of errors related to unneeded usings after these changes. I did not remove the unneeded usings in this PR to not bloat it and will create a second PR once this hits the main to fix the issues.

File: src/Quartz.Tests.Integration/Impl/SmokeTestPerformer.cs [hunk 1]
```diff
@@ -516,4 +513,24 @@ protected override void DeserializeFields(CustomCalendar calendar, JObject sourc
     {
         calendar.SomeCustomProperty = source["SomeCustomProperty"]!.Value<bool>();
     }
+}
+
+internal sealed class CustomSystemTextJosnCalendarSerializer : Serialization.Json.Calendars.CalendarSerializer<CustomCalendar>
```
Comment (lahma) [inline src/Quartz.Tests.Integration/Impl/SmokeTestPerformer.cs:518]: `Josn` 😉

**Review by lahma — APPROVED** [review]
Awesome, thank you!

File: src/Quartz.Tests.Integration/Impl/SmokeTestPerformer.cs (hunk 1 above)
Comment (VilleHakli) [reply src/Quartz.Tests.Integration/Impl/SmokeTestPerformer.cs:518]: That's the advanced version of JSON :)

**Pull request merged**